### PR TITLE
(#1998) Add authorization header to Get-WebFile

### LINKED ISSUE #1998: Get-WebFile does not handle authorization headers
Passing authorization headers through `Install-ChocolateyPackage` does not attach them to `Get-WebFile` downstream properly.

### PR DESCRIPTION
**Pull request by steviecoaster**
Fixes #1998 

* Reworks header manipulation to use $key instead of `GetEnumerator()` method

### CODE REVIEW

File: src/chocolatey.resources/helpers/functions/Get-WebFile.ps1 [hunk 1]
```diff
@@ -177,17 +177,16 @@ param(
     $req.UserAgent = $userAgent
   }
 
-  if ($options.Headers.Count -gt 0) {
+  if($options.Headers.Count -gt 0){
     Write-Debug "Setting custom headers"
-    foreach ($item in $options.Headers.GetEnumerator()) {
-      $uri = (new-object system.uri $url)
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/functions/Get-WebFile.ps1]: @steviecoaster looks like you cut this one out by accident. It's used when setting the cookies so you'll probably need that. 🙂

File: src/chocolatey.resources/helpers/functions/Get-WebFile.ps1 [hunk 2]
```diff
@@ -177,17 +177,17 @@ param(
...
-      Write-Debug($item.Key + ':' + $item.Value)
-      switch ($item.Key) {
-        'Accept' {$req.Accept = $item.Value}
-        'Cookie' {$req.CookieContainer.SetCookies($uri, $item.Value)}
-        'Referer' {$req.Referer = $item.Value}
-        'User-Agent' {$req.UserAgent = $item.Value}
-        Default {$req.Headers.Add($item.Key, $item.Value)}
+    foreach($key in $options.headers.keys){
+      $uri = (New-Object sysetem.uri $url)
```
Comment (vexx32) [inline src/chocolatey.resources/helpers/functions/Get-WebFile.ps1]:
```suggestion
      $uri = New-Object -TypeName System.Uri $url

```

File: src/chocolatey.resources/helpers/functions/Get-WebFile.ps1 (hunk 1 above)
Comment (steviecoaster) [reply src/chocolatey.resources/helpers/functions/Get-WebFile.ps1]: Fixed

File: src/chocolatey.resources/helpers/functions/Get-WebFile.ps1 (hunk 2 above)
Comment (steviecoaster) [reply src/chocolatey.resources/helpers/functions/Get-WebFile.ps1]: If we are being pedantic I'll just use:

`$uri = [System.Uri]$url`

;)

Comment (steviecoaster) [reply src/chocolatey.resources/helpers/functions/Get-WebFile.ps1]: Fixed.

File: src/chocolatey.resources/helpers/functions/Get-WebFile.ps1 [hunk 3]
```diff
@@ -177,17 +177,17 @@ param(
...
-        'User-Agent' {$req.UserAgent = $item.Value}
-        Default {$req.Headers.Add($item.Key, $item.Value)}
+    foreach($key in $options.headers.keys){
+      $uri = (New-Object -Typename system.uri $url)
+      switch($key){
+        'Accept' {$req.Accept = $options.headers.$key}
+        'Cookie' {$req.CookieContainer.SetCookies($uri,$options.headers.$key)}
+        'Referer' {$req.Referer = $options.headers.$key}
+        'User-Agent' {$req.UserAgent = $options.headers.$key}
+        'Authorization' {$re.Autorization = $options.headers.$key}
```
Comment (ferventcoder) [inline src/chocolatey.resources/helpers/functions/Get-WebFile.ps1:189]: accidental misspelling?

**Comment by gep13** [comment]
@steviecoaster this has been rebased and merged here: https://github.com/chocolatey/choco/commit/5580558c52c423f578a9a68d13e19e42161c5dcd

I wasn't able to push this changes to your fork, as it looks like it has been deleted, but your commit has still been correctly attributed.

I will go ahead and close this PR, as it is no longer required.

**Comment by gep13** [comment]
@steviecoaster thanks for your help in getting this fixed up!

**Pull request closed without merging**